bin/i18n/sync-codeorg-out

```diff
@@ -82,4 +83,105 @@ def rebuild_blockly_js_files
+    return value.scan(link_or_image_re)
+  elsif value.instance_of? Array
+    return value.join(' ').scan(link_or_image_re)
+  elsif value.instance_of? Hash
+    values = value.keys.sort.map do |key|
+      get_links_and_images(value[key])
+    end
+    return values.flatten
+  end
+end
+
+def check_for_bad_translations
+  categories_to_check = %w(
+    instructions
+    markdown_instructions
+    failure_message_overrides
+    authored_hints
+    callouts
+  )
+
+  outputfile = Tempfile.new('sync-codeorg-out')
+
+  Languages.get_locale.each do |prop|
+    locale = prop[:locale_s]
+
+    # don't check english, as it is our base language
+    next if locale == 'en-US'
+
+    # also don't check Italian; they have a lot of custom translation
+    # work which we whitelist
+    next if locale == 'it-IT'
+
+    print "validating #{locale} ..."
+
+    mismatch_found = false
+
+    categories_to_check.each do |category|
+      locale_file = "dashboard/config/locales/#{category}.#{locale}.yml"
+      source_file = "dashboard/config/locales/#{category}.en.yml"
+
+      # this absurd little dance is necessary because of the format of
+      # each of these yaml files; they each have a hash with a single
+      # key representing the two-character version of the locale whose
+      # value is a hash with the single key 'data' whose value is a hash
+      # with a single key representing the category name whose value is
+      # (finally) the data we actually want.
+      source_data = YAML.load(File.open(source_file)).values.first.values.first.values.first
+      locale_data = YAML.load(File.open(locale_file)).values.first.values.first.values.first
+
+      if source_data.keys != locale_data.keys
+        outputfile.write "mismatching keys in #{locale_file}"
+        mismatch_found = true
+      end
+
+      mismatching_keys = locale_data.keys.select do |key|
+        get_links_and_images(source_data[key]) != get_links_and_images(locale_data[key])
+      end
+
+      unless mismatching_keys.empty?
+        mismatch_found = true
+        outputfile.write("mismatching values in #{locale_file}\n")
+        outputfile.write("and therefore also in i18n/locales/#{locale}/dashboard/#{category}.yml:\n")
+        mismatching_keys.each do |key|
+          outputfile.write("\t#{key}\n")
+          outputfile.write("\t\ten:\n")
+          get_links_and_images(source_data[key]).each do |x|
+            outputfile.write("\t\t\t#{x}\n")
+          end
+          outputfile.write("\t\t#{locale}:\n")
+          get_links_and_images(locale_data[key]).each do |x|
+            outputfile.write("\t\t\t#{x}\n")
+          end
+        end
+        outputfile.write("\n")
+      end
+    end
+
+    if mismatch_found
+      # if we find at least one mismatch, remove the finalizer from the
+      # tempfile that would otherwise delete it when this script exits
+      ObjectSpace.undefine_finalizer(outputfile)
+      puts " MISMATCH FOUND! output in #{outputfile.path}"
```

Comment:
I'm not able to run this so I'm not sure what this exported file looks like. To be safe (especially if other people run the i18n scripts) do we want to gitignore this file?